### ISSUE TITLE
Compiles without index specified.

### ISSUE THREAD

**Issue by PatrickGTR** [pawn-lang/compiler#529]
### Minimal complete verifiable example (MCVE):

```c
static 
    _reproduce[5];

hook OnGameModeInit() {

    // Index for reproduce isn't specified, compiler should complain!
    // instead it prints the value of the first index of.
    printf("%i", _reproduce);
    _reproduce[0] += 2;
    printf("%i", _reproduce);
    _reproduce[1] += 4;
    printf("%i", _reproduce);

    printf("with index %i[0] %i[1]", _reproduce[0], _reproduce[1]);
    return 1; 
}
```
### output 
```c
0
2
2
with index 2[0] 4[1]
```

### Workspace Information:

* Compiler version: < 3.10.10
* Command line arguments provided (or sampctl version): 
```json
{

	"local": true,
	"builds": 
	{
		"version": "3.10.10",
		"args": [
			"-;+",
			"-(+",
			"-d3",
			"-E"
		]
	}
}
```
* Operating System: Windows 10 Pro

**Comment by Y-Less** [comment]
That's not a bug.  `printf` can take numbers and strings.  Passing `_reproduce` just looks like you're passing a string.  To the compiler `"%i"` is just an arbitrary string with no meaning.
